### ISSUE TITLE
tar command failed

### ISSUE THREAD

**Issue by RajaSrinivasan** [alire-project/alire#726]
Environment: Windows

Using git bash

I executed bin/alr printenv
and answered yes to installing msys2

Warning: tar failed: Command ["tar", "-x", "-f", "C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ada_toml_0.1.0_2235adad\v0.1.tar.gz"] exited with code 128 and output: /usr/bin/tar: Cannot connect to C: resolve failed
ERROR: Deployment of source archive v0.1.tar.gz at https://github.com/pmderodat/ada-toml/archive/v0.1.tar.gz with hash sha512:2235adad4908c0b4e890bf314fca8fc5c374564b2ca5dd9844090b9845de89d6401a431344124d63251a1ff4b5fad5ea2d13fb21b996019289106f7f2231c727 to C:\Users\Raja
S\Projects\alire\alire\cache\dependencies\ada_toml_0.1.0_2235adad failed

**Comment by Fabien-Chouteau** [comment]
Hello @RajaSrinivasan ,
Tar is a little bit of an issue on Windows unfortunately. I think that tar from git bash is used here and it doesn't work the same as the one provided in msys2.

Which version of Alire are you using (`$alr version`)?

**Comment by RajaSrinivasan** [comment]
Hi Fabien - here is the output.

Alr version: 1.1.0-dev+0f603c29
Alire Library version: 1.1.0-dev+0f603c29
alr status is (valid) (0 releases indexed) (loaded in 0.713s)
config folder is C:\Users\RajaS\.config\alire
source folder is C:\Users\RajaS\.config\alire\alire
interaction flags are: force:FALSE not-interactive:FALSE
alr root is empty
alr root detection has settled on path: C:\Users\RajaS\Projects\alire
alr is finding 4 GPR project files
alr session state is [OUTSIDE]
alr compiled on [2021-04-25 06:11:44] with GNAT version [Community 2020 (2020042
9-93)]
platform fingerprint: Windows Bits_64 Msys2
platform properties: MSYS2 WINDOWS NATIVE USER BITS_64
community index required branch: stable-1.0

**Comment by Fabien-Chouteau** [comment]
Thank you @RajaSrinivasan,

Can you please try again the command that fails but with the `-vv -d` options: `alr -vv -d printenv`?

**Comment by RajaSrinivasan** [comment]
Using -d -vv is rather verbose. captured the tail end. it would appear that this time, it was successful. 

Anything else i can check. i am interested in some GNATCOLL packages and some aws packages, i will let you know if there are issues.

thanks, srini

-> Using default value for configuration 'update-manually-only': 'FALSE'
-->> Workspace is being checked and loaded for the first time
-->> Looking for alire metadata at: C:\Users\RajaS\Projects\alire
-->> Loading release alr=0.0.0
-->> Valid root found at C:\Users\RajaS\Projects\alire
-->> Loading release alr=0.0.0
-->> Checked valid manifest at C:\Users\RajaS\Projects\alire\alire.toml
-->> Reading persistent contents from C:\Users\RajaS\Projects\alire\alire.lock
-->> Reading solution from TOML...
-->> Loading release aaa=0.0.0
-->> Loading release ada_toml=0.0.0
-->> Loading release ajunitgen=0.0.0
-->> Loading release ansiada=0.0.0
-->> Loading release curl=0.0.0
-->> Loading release gnatcoll=0.0.0
-->> Loading release libgpr=0.0.0
-->> Loading release minirest=0.0.0
-->> Loading release semantic_versioning=0.0.0
-->> Loading release simple_logging=0.0.0
-->> Loading release spdx=0.0.0
-->> Loading release uri_ada=0.0.0
-->> Loading release xml_ez_out=0.0.0
-->> Loading release xmlada=0.0.0
-->> Lockfile at C:\Users\RajaS\Projects\alire\alire.lock is valid
-->> Reading persistent contents from C:\Users\RajaS\Projects\alire\alire.lock
-->> Reading solution from TOML...
-->> Loading release aaa=0.0.0
-->> Loading release ada_toml=0.0.0
-->> Loading release ajunitgen=0.0.0
-->> Loading release ansiada=0.0.0
-->> Loading release curl=0.0.0
-->> Loading release gnatcoll=0.0.0
-->> Loading release libgpr=0.0.0
-->> Loading release minirest=0.0.0
-->> Loading release semantic_versioning=0.0.0
-->> Loading release simple_logging=0.0.0
-->> Loading release spdx=0.0.0
-->> Loading release uri_ada=0.0.0
-->> Loading release xml_ez_out=0.0.0
-->> Loading release xmlada=0.0.0
-> Index already loaded, loading skipped
-->> Reading persistent contents from C:\Users\RajaS\Projects\alire\alire.lock
-->> Reading solution from TOML...
-->> Loading release aaa=0.0.0
-->> Loading release ada_toml=0.0.0
-->> Loading release ajunitgen=0.0.0
-->> Loading release ansiada=0.0.0
-->> Loading release curl=0.0.0
-->> Loading release gnatcoll=0.0.0
-->> Loading release libgpr=0.0.0
-->> Loading release minirest=0.0.0
-->> Loading release semantic_versioning=0.0.0
-->> Loading release simple_logging=0.0.0
-->> Loading release spdx=0.0.0
-->> Loading release uri_ada=0.0.0
-->> Loading release xml_ez_out=0.0.0
-->> Loading release xmlada=0.0.0
-> Env: msys2 appends 'C:\msys64\mingw64\lib' to 'LIBRARY_PATH'
-> Env: ansiada (gpr ext) sets 'On_Demand' to 'ANSIADA_BUILD_MODE'
-> Env: ada_toml (gpr ext) sets 'static' to 'LIBRARY_TYPE'
-> Env: ada_toml (gpr ext) sets 'prod' to 'BUILD_MODE'
-> Env: msys2 appends 'C:\msys64\mingw64\bin' to 'PATH'
-->> Skipping identical key value: PATH=C:\msys64\usr\bin;C:\msys64\usr\local\bin;C:\msys64\mingw64\bin;C:\msys64\usr\local\bin;C:\msys64\usr\bin;C:\msys64\usr\bin;C:\Windows\System32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\msys64\usr\bin\site_perl;C:\msys64\usr\bin\vendor_perl;C:\msys64\usr\bin\core_perl
-> Env: msys2 appends 'C:\msys64\usr\bin' to 'PATH'
-->> Skipping identical key value: PATH=C:\msys64\usr\bin;C:\msys64\usr\local\bin;C:\msys64\mingw64\bin;C:\msys64\usr\local\bin;C:\msys64\usr\bin;C:\msys64\usr\bin;C:\Windows\System32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\msys64\usr\bin\site_perl;C:\msys64\usr\bin\vendor_perl;C:\msys64\usr\bin\core_perl
-> Env: msys2 appends 'C:\msys64\usr\local\bin' to 'PATH'
-->> Skipping identical key value: PATH=C:\msys64\usr\bin;C:\msys64\usr\local\bin;C:\msys64\mingw64\bin;C:\msys64\usr\local\bin;C:\msys64\usr\bin;C:\msys64\usr\bin;C:\Windows\System32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\msys64\usr\bin\site_perl;C:\msys64\usr\bin\vendor_perl;C:\msys64\usr\bin\core_perl
-> Env: gnatcoll (gpr ext) sets 'windows' to 'GNATCOLL_OS'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\aaa_0.2.1_103f0d19' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ada_toml_0.1.0_2235adad' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ajunitgen_1.0.1_080ca963' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ansiada_0.1.0_27a89150' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\gnatcoll_21.0.0_6c0439a3' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\libgpr_21.0.0_49263e36\gpr' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\minirest_0.1.0_272ecd15' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\semantic_versioning_2.0.0_2f76a998' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\simple_logging_1.2.0_506f0bb3' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\spdx_0.2.0_2df9b118' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\uri_ada_1.0.0_84a01e5c' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xml_ez_out_1.6.0_c90d08dc' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\distrib' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\dom' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\input_sources' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\sax' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\schema' to 'GPR_PROJECT_PATH'
-> Env: crates appends 'C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\unicode' to 'GPR_PROJECT_PATH'
-> Env: Alire sets 'True' to 'ALIRE'
-> Env: gnatcoll (gpr ext) sets '21.0.0' to 'GNATCOLL_VERSION'
-> Env: msys2 appends 'C:\msys64\mingw64\include' to 'C_INCLUDE_PATH'
export ALIRE="True"
export ANSIADA_BUILD_MODE="On_Demand"
export BUILD_MODE="prod"
export C_INCLUDE_PATH="C:\msys64\mingw64\include"
export GNATCOLL_OS="windows"
export GNATCOLL_VERSION="21.0.0"
export GPR_PROJECT_PATH="C:\Users\RajaS\Projects\alire;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\aaa_0.2.1_103f0d19;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ada_toml_0.1.0_2235adad;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ajunitgen_1.0.1_080ca963;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\ansiada_0.1.0_27a89150;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\gnatcoll_21.0.0_6c0439a3;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\libgpr_21.0.0_49263e36\gpr;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\minirest_0.1.0_272ecd15;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\semantic_versioning_2.0.0_2f76a998;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\simple_logging_1.2.0_506f0bb3;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\spdx_0.2.0_2df9b118;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\uri_ada_1.0.0_84a01e5c;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xml_ez_out_1.6.0_c90d08dc;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\distrib;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\dom;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\input_sources;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\sax;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\schema;C:\Users\RajaS\Projects\alire\alire\cache\dependencies\xmlada_21.0.0_836d1890\unicode"
export LIBRARY_PATH="C:\msys64\mingw64\lib"
export LIBRARY_TYPE="static"
export PATH="C:\msys64\usr\bin;C:\msys64\usr\local\bin;C:\msys64\mingw64\bin;C:\msys64\usr\local\bin;C:\msys64\usr\bin;C:\msys64\usr\bin;C:\Windows\System32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\msys64\usr\bin\site_perl;C:\msys64\usr\bin\vendor_perl;C:\msys64\usr\bin\core_perl"
-> alr printenv done

**Comment by Fabien-Chouteau** [comment]
I am sorry `printenv` doesn't use the tar. What command did you use in the first place?

**Comment by RajaSrinivasan** [comment]
Seems like the issue resolved itself. I wish I knew how.

**Comment by sanyaade-teachings** [comment]
Hi All,
This issue is still there!

Setup alire on Windows 7 64Bit along with Gnat studio. Everything working but when I tried out alire with command:
     alr get hello
I got the tar error message below:
![image](https://user-images.githubusercontent.com/4993211/133888429-a1513a84-5a2d-402d-926d-f55520052bc6.png)

Basically alire was looking for tar archive manager to decompress some of the downloads.

I have 7zip installed, added to path and shell but alr get hello fails again.

I went through steps listed above ==> alr  -vv  -d  printenv
The dialogue messages is as shown below:
![image](https://user-images.githubusercontent.com/4993211/133888736-5f8b0c47-164b-43db-a2fa-ac305c3fea23.png)

I need to get this issues resolved as I want my students to use alire during the course/teaching sessions.
Please help!

Many thanks!

God blesses!!!

Best regards,
Sanyaade

**Comment by sanyaade-teachings** [comment]
Hi There,

I got this sorted out. It was due to a previously installed tar package for AVR on my system. I got rid of that one and now Alire run, download and install new tar packages and also completed installation of the hello project.  :+1: 

![image](https://user-images.githubusercontent.com/4993211/133896270-07170f94-997c-435c-80a6-2ea75a15767c.png)

I am doing a handout on the use of Alire for my students, so I may shout out when needed here!

Many thanks!

God blesses!!!

Best regards,
Sanyaade
